### PR TITLE
Clang-tidy for steam interop source code.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

**Comment by sergio-nsk** [comment]
Why is interop.c still C while all other files are C++?

File: steamapi.cc [hunk 1]
```diff
@@ -16,114 +16,109 @@
...
-    } else
-        GlobalSteamIsInitialized = false;
-
-    return GlobalSteamIsInitialized;
+    if (SteamAPI_IsSteamRunning())
+        g_steam_is_initialized = SteamAPI_Init();
+    else
+        g_steam_is_initialized = false;
+
+    return g_steam_is_initialized;
```
Comment (sergio-nsk) [inline steamapi.cc]: Unnamed namespace?

File: steamapi.cc [hunk 2]
```diff
@@ -16,114 +16,109 @@
...
 
     SteamAPI_RunCallbacks();
     return true;
 }
 
-bool SteamAPI_Invoke(void *SteamAPIContext, echandle MethodDictionaryHandle, echandle ReturnDictionaryHandle) {
+bool SteamAPI_Invoke(void *handle, echandle method_dictionary_handle, echandle return_dictionary_handle) {
     // EVERYTHING is marshaled in AND out as a JSON string, use any type supported by JSON and
     // it should marshal ok.
+    bool ret_val = false;
```
Comment (sergio-nsk) [inline steamapi.cc]: `ret`?

File: steamapi.cc [hunk 3]
```diff
@@ -16,114 +16,109 @@
...
+    int32_t return_value = false;
```
Comment (sergio-nsk) [inline steamapi.cc]: `bool`?

File: interop.c [hunk 4]
```diff
@@ -12,41 +12,42 @@
...
 
-bool Interop_CreateInstance(const char *TypeName, char *InstanceId, int32_t InstanceIdLength, void *ExecuteUserPtr,
-                            Interop_ExecuteCallback Execute, Interop_InvokeInstanceCallback *InvokeInstance,
-                            Interop_ReleaseInstanceCallback *ReleaseInstance,
-                            Interop_ProcessInstanceCallback *ProcessInstance, void **UserPtr) {
-    *InvokeInstance = NULL;
-    *ReleaseInstance = NULL;
-    *ProcessInstance = NULL;
-    *UserPtr = NULL;
+bool Interop_CreateInstance(const char *type_name, char *instance_id, int32_t instance_id_length,
```
Comment (sergio-nsk) [inline interop.c]: `max_instance_id` for consistency with the other API?

**Comment by nmoinvaz** [comment]
There wasn't any need to change it to .cc because the files are so simple.

File: steamapps.cc [hunk 5]
```diff
@@ -17,186 +17,126 @@
...
-    return SteamApps()->GetDLCCount() != 0;
-}
+namespace {
 
-static bool SteamApps_GetDLCData(int32_t Index, int32_t *AppId, int32_t *Available, char *Name, int32_t MaxName) {
-    uint32_t AppIdInt = 0;
-    bool AvailableInt = false;
+bool GetDLCData(int32_t index, int32_t *app_id, bool *available, char *name, int32_t max_name) {
+    uint32_t app_id_int = 0;
+    bool available_int = false;
```
Comment (sergio-nsk) [inline steamapps.cc]: `available` became bool, `vailable_int` became unnecessary.

File: steamapps.cc [hunk 6]
```diff
@@ -17,186 +17,126 @@
...
     // it should marshal ok.
 
-    echandle ItemHandle = nullptr;
-    bool RetVal = false;
-    int32_t ReturnValue = false;
-    int32_t Value32 = 0;
-    bool ValueBool = false;
-    const char *Method = nullptr;
-    char *ValueString = nullptr;
+    bool ret_val = false;
```
Comment (sergio-nsk) [inline steamapps.cc]: `ret`?

File: steamapps.cc [hunk 7]
```diff
@@ -17,186 +17,126 @@
...
+    int32_t return_value = false;
```
Comment (sergio-nsk) [inline steamapps.cc]: `bool`?

File: steamfriends.cc [hunk 8]
```diff
@@ -14,145 +14,111 @@
...
-static bool SteamFriends_GetFriendRelationship(uint64_t FriendId, int32_t *Relationship) {
-    *Relationship = SteamFriends()->GetFriendRelationship(CSteamID((uint64)FriendId));
-    return true;
-}
+namespace {
 
-static bool SteamFriends_GetSmallFriendAvatar(uint64_t FriendId, int32_t *SmallFriendAvatar) {
-    *SmallFriendAvatar = SteamFriends()->GetSmallFriendAvatar(CSteamID((uint64)FriendId));
+bool GetFriendPersonaName(uint64_t friend_id, char *persona_name, int32_t max_persona_name) {
+    strncpy(persona_name, SteamFriends()->GetFriendPersonaName(CSteamID(static_cast<uint64>(friend_id))),
```
Comment (sergio-nsk) [inline steamfriends.cc:20]: `uint64_t`?

File: steamapi.cc [hunk 9]
```diff
@@ -16,114 +16,109 @@
...
-    } else if (strcmp(Method, "setAppId") == 0) {
-        RetVal = IDictionary_GetInt32ByKey(MethodDictionaryHandle, "id", &Value32);
-        if (RetVal)
-            ReturnValue = SteamAPI_SetAppId(Value32);
-        IDictionary_AddBoolean(ReturnDictionaryHandle, "returnValue", ReturnValue, &ItemHandle);
+    if (strcmp(method, "initialize") == 0) {
+        return_value = SteamAPI_InitializeInt();
+        ret_val = IDictionary_AddBoolean(return_dictionary_handle, "returnValue", return_value, nullptr);
+    } else if (strcmp(method, "isSteamRunning") == 0) {
+        return_value = (int32)SteamAPI_IsSteamRunning();
```
Comment (sergio-nsk) [inline steamapi.cc]:
```suggestion
        return_value = SteamAPI_IsSteamRunning();
```

File: steamapi.cc [hunk 10]
```diff
@@ -16,114 +16,109 @@
...
+        ret_val = IDictionary_AddBoolean(return_dictionary_handle, "returnValue", return_value, nullptr);
+    } else if (strcmp(method, "isInitialized") == 0) {
+        return_value = SteamAPI_IsInitialized();
+        ret_val = IDictionary_AddBoolean(return_dictionary_handle, "returnValue", return_value, nullptr);
+    } else if (strcmp(method, "restartAppIfNecessary") == 0) {
+        int32_t value = 0;
+        ret_val = IDictionary_GetInt32ByKey(method_dictionary_handle, "id", &value);
+        if (ret_val)
+            return_value = (int32)SteamAPI_RestartAppIfNecessary(value);
```
Comment (sergio-nsk) [inline steamapi.cc]:
```suggestion
            return_value = SteamAPI_RestartAppIfNecessary(value);
```

File: steamapps.cc [hunk 11]
```diff
@@ -17,186 +17,126 @@
...
+        ret_val = IDictionary_GetInt32ByKey(method_dictionary_handle, "id", &app_id);
+        if (ret_val)
+            return_value = SteamApps()->BIsAppInstalled(app_id) != 0;
+        IDictionary_AddBoolean(return_dictionary_handle, "returnValue", return_value, nullptr);
+    } else if (strcmp(method, "getAppInstallDir") == 0) {
+        int32_t app_id = 0;
+        char install_dir[320] = {0};
+        ret_val = IDictionary_GetInt32ByKey(method_dictionary_handle, "id", &app_id);
+        if (ret_val)
+            return_value = SteamApps()->GetAppInstallDir(app_id, install_dir, sizeof(install_dir)) != 0;
```
Comment (sergio-nsk) [inline steamapps.cc]: `return_value` is unused, maybe `ret_val`/`ret`?

File: steamuser.cc [hunk 12]
```diff
@@ -16,212 +16,159 @@
...
-    SteamUser()->StopVoiceRecording();
-    return true;
+    if (response->m_eResult == k_EResultOK)
+        successful = true;
+    if (response->m_hAuthTicket == this->Ticket) {
+        char hex_ticket[4096];
+        String_ConvertToHex(this->TicketData, this->TicketSize, true, hex_ticket, sizeof(hex_ticket));
+        NotificationCenter_FireAfterDelayWithJSON(
+            "SteamUser", "AuthSessionTicketResponse", Class_InstanceId((SteamUserStruct *)this->User), 0,
+            "{ \"ticket\": %d, \"ticketData\": \"%s\", \"successful\": %s, \"errorCode\": %d }", this->Ticket,
```
Comment (sergio-nsk) [inline steamuser.cc]: `this->` is unnecessary.

File: steamuser.cc [hunk 13]
```diff
@@ -16,212 +16,159 @@
 class UserResults {
   public:
     CCallbackManual<UserResults, GetAuthSessionTicketResponse_t, false> AuthSessionTicket;
-    void OnAuthSessionTicketResponse(GetAuthSessionTicketResponse_t *Response);
 
     void *User = nullptr;
+
     HAuthTicket Ticket = k_HAuthTicketInvalid;
     uint8_t TicketData[2048] = {0};
     uint32_t TicketSize = 0;
```
Comment (sergio-nsk) [inline steamuser.cc]: Member case?

File: steamuser.cc [hunk 14]
```diff
@@ -16,212 +16,159 @@
...
+};
 
 /********************************************************************/
 
-static SteamUserStruct *GlobalSteamUser = nullptr;
+namespace {
 
-/********************************************************************/
-// Callback functions
+struct SteamUserStruct {
```
Comment (sergio-nsk) [inline steamuser.cc:33]: `Struct` does not seem necessary. Can be `struct SteamUser : ClassStruct {`.

File: steamuser.cc [hunk 15]
```diff
@@ -16,212 +16,159 @@
...
-static bool SteamUser_StopVoiceRecording(void) {
-    SteamUser()->StopVoiceRecording();
-    return true;
+    if (response->m_eResult == k_EResultOK)
+        successful = true;
+    if (response->m_hAuthTicket == this->Ticket) {
+        char hex_ticket[4096];
+        String_ConvertToHex(this->TicketData, this->TicketSize, true, hex_ticket, sizeof(hex_ticket));
+        NotificationCenter_FireAfterDelayWithJSON(
+            "SteamUser", "AuthSessionTicketResponse", Class_InstanceId((SteamUserStruct *)this->User), 0,
```
Comment (sergio-nsk) [inline steamuser.cc]: `static_cast<SteamUser *>(user)->InstanceId` like in other usages.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]
Some changes like type names, `ret_val` -> `ret`, `int32_t` -> `bool` are desired and repetitive in each .cc file. Other changes may be considered as refactoring can be skipped in this PR.

File: interop.c (hunk 4 above)
Comment (nmoinvaz) [reply interop.c]: Hmm, we have used `instance_id_length` for other `Interop_CreateInstance` calls in client. Not sure I really want to update all those..

Comment (sergio-nsk) [reply interop.c]: I see that in the callback
```c++
typedef bool (*Interop_CreateInstanceCallback)(const char *type_name, char *instance_id, int32_t max_instance_id,
```
and implementations are like
```c++
<...>_CreateInstance(const char *type_name, char *instance_id, int32_t instance_id_length,
```
Disorder.

File: steamfriends.cc (hunk 8 above)
Comment (nmoinvaz) [reply steamfriends.cc:20]: `uint64` is from Steam API. It is defined differently than `uint64_t`.

File: steamuser.cc (hunk 13 above)
Comment (nmoinvaz) [reply steamuser.cc]: What are you referring to?

File: steamuser.cc (hunk 14 above)
Comment (nmoinvaz) [reply steamuser.cc:33]: I don't think so. It throws a bunch of errors about `Class` not existing in the structure.

File: steamuser.cc (hunk 15 above)
Comment (nmoinvaz) [reply steamuser.cc]: Fixed.

File: steamuser.cc (hunk 12 above)
Comment (nmoinvaz) [reply steamuser.cc]: Fixed.

File: steamapps.cc (hunk 11 above)
Comment (nmoinvaz) [reply steamapps.cc]: Fixed.

File: steamapi.cc (hunk 10 above)
Comment (nmoinvaz) [reply steamapi.cc]: Fixed.

File: steamapi.cc (hunk 9 above)
Comment (nmoinvaz) [reply steamapi.cc]: Fixed.

File: steamapps.cc (hunk 7 above)
Comment (nmoinvaz) [reply steamapps.cc]: Fixed.

File: steamapps.cc (hunk 6 above)
Comment (nmoinvaz) [reply steamapps.cc]: Fixed.

File: steamapps.cc (hunk 5 above)
Comment (nmoinvaz) [reply steamapps.cc]: Fixed.

File: steamapi.cc (hunk 3 above)
Comment (nmoinvaz) [reply steamapi.cc]: Fixed.

File: steamapi.cc (hunk 2 above)
Comment (nmoinvaz) [reply steamapi.cc]: Fixed.

File: steamapi.cc (hunk 1 above)
Comment (nmoinvaz) [reply steamapi.cc]: Fixed.

File: steamuser.cc (hunk 13 above)
Comment (sergio-nsk) [reply steamuser.cc]: Character cases in member names are inconsistent. `SteamUserStruct::results` and `UserResults::User`.

File: interop.c (hunk 4 above)
Comment (nmoinvaz) [reply interop.c]: Ok, I will do it and assign the other PRs to you.

File: steamuser.cc (hunk 14 above)
Comment (sergio-nsk) [reply steamuser.cc:33]: How so? I saw only a couple of `Class_InstanceId` usages that could be easy replaced with `self->InstanceId`.

File: steamuser.cc (hunk 13 above)
Comment (nmoinvaz) [reply steamuser.cc]: In the C++ class those variables are public. Are you wanting me to rename them to lowercase first char? Struct has no concept of public/private, so I made it lowercase. Do you want it uppercase? What do you want?

File: steamuser.cc (hunk 14 above)
Comment (nmoinvaz) [reply steamuser.cc:33]: It is too much refactoring. I don't feel safe about the change not causing problems.

File: steamuser.cc (hunk 13 above)
Comment (sergio-nsk) [reply steamuser.cc]: Structs are classes. Only their default accesses differ.
As for the code style, public member variables can be `var_name` or `var_name_`.
```c++
struct B {
  int b;
};
// is
class B {
 public:
  int b;
};
// and vice versa
class A : public B {
  int a_;
};
// is
struct A : B {
 private:
  int a_;
};

```

Comment (nmoinvaz) [reply steamuser.cc]: Ok, I have made them all lowercase for you.

**Review by sergio-nsk — APPROVED** [review]
-237 lines. Can be even better cleaned up.

**Pull request merged**